update docker helper to be current (#882)
medic/cht-docs: content/en/apps/guides/hosting/4.x/app-developer.md

This script brings a lot of benefits with it:
* You only have to download one bash script
* All compose files and images will be downloaded automatically for you
* All networks, storage volumes and containers will be created 
* A valid TLS certificate will be installed, allowing you to easily test on with CHT Android natively on a mobile device
* An unused port is automatically chosen for you when creating a new project.  No more manually looking at your existing `.env` files!

### Installing

To get started using it:
1. Clone the [CHT Core](https://github.com/medic/cht-core/) repo
2. When you want to check for updates, just run `git pull orign` in the `cht-core` directory.

If you want a more stand-alone version, you can `curl` the bash script directly, but you can't use `git` to easily update it then:

```shell
curl https://raw.githubusercontent.com/medic/cht-core/master/scripts/docker-helper-4.x/cht-docker-compose.sh
```

### Usage

> Always run the script from directory where it lives.  If you launch it from further up the directory three, relative paths will fail:

Reviewer: ```suggestion
Always run the script from the directory where it lives.  If you launch it from a different directory, relative paths will fail:
```
typo

| Do | Don't |
|---|---|
| `./cht-docker-compose.sh`|`./docker-helper-4.x/cht-docker-compose.sh`|

#### Launching

Run the script with:

```
./cht-docker-compose.sh
```

The first time you run, you will be prompted to create a new project.  Here's what that looks like:

```shell
./cht-docker-compose.sh
Would you like to initialize a new project [y/N]? y
How do you want to name the project? 4 OH The First
Downloading compose files ... done 

Creating network "4_oh_the_first-cht-net" with the default driver
Creating my_first_cht_project-dir_cht-upgrade-service_1 ... done
Starting project "4_oh_the_first". First run takes a while. Will try for up to five minutes........

 -------------------------------------------------------- 

  Success! "4_oh_the_first" is set up:

    https://127-0-0-1.my.local-ip.co:10444/ (CHT)
    https://127-0-0-1.my.local-ip.co:10444/_utils/ (Fauxton)

    Login: medic
    Password: password

 -------------------------------------------------------- 

>  Have a great day! 

Reviewer: ```suggestion
Start existing project
    ./cht-docker-compose.sh ENV-FILE.env

Stop and keep project:
    ./cht-docker-compose.sh ENV-FILE.env stop

Stop and destroy all project data:
    ./cht-docker-compose.sh ENV-FILE.env destroy

https://docs.communityhealthtoolkit.org/apps/guides/hosting/4.x/app-developer/

 Have a great day! 
```

Not sure we actually need to include all of this, but otherwise we should probably drop the "Have a great day!" part and just stick with the instance info here...

```

If you have many existing projects, you can specify them to launch them directly. If you had a project called `4_oh_the_first` you would run:

```shell
./cht-docker-compose.sh 4_oh_the_first.env
```

#### Stopping

When you're done with a project, it's good to stop all the containers to reduce load on your computer.  Do this by specifying the project and the `down` command. This command will simply stop the active Docker containers, and not delete any data. Using our existing example `4_oh_the_first` project, you would call:

```shell
./cht-docker-compose.sh 4_oh_the_first.env down
```

#### Destroying

When you want to **permanently delete all files and all data** for a project, specify the project and the `destroy` command. Using our existing example `4_oh_the_first` project, you would call:

```shell
./cht-docker-compose.sh 4_oh_the_first.env destroy
```

> Be sure you want to do this, there is no "are you sure?" prompt and it will delete all your date.

Reviewer: ```suggestion
Be sure you want to do this, there is no "are you sure?" prompt and it will delete all your data.
```

Also note that this command will use the `sudo` command when deleting the CouchDB data, so it may prompt for your password.

### File locations

The bash script keeps files in two places:

* **`*.env` files** -  the same directory as the bash script. 

> * **`~/medic/cht-docker/` files** - in your home directory, a sub-directory is created for each project.  Within each project directory, a `compose` has all the two compose files and the `couch` directory has the CouchDB datafiles.

Reviewer: ```suggestion
* **`~/medic/cht-docker/` files** - in your home directory, a sub-directory is created for each project.  Within each project directory, a `compose` directory has the two compose files and the `couch` directory has the CouchDB datafiles.
```

* All compose files and images will be downloaded automatically for you
* All networks, storage volumes and containers will be created 
* A valid TLS certificate will be installed, allowing you to easily test on with CHT Android natively on a mobile device
* An unused port is automatically chosen for you when creating a new project.  No more manually looking at your existing `.env` files!

### Installing

To get started using it:
1. Clone the [CHT Core](https://github.com/medic/cht-core/) repo

> 4. When you want to check for updates, just run `git pull orign` in the `cht-core` directory.

Reviewer: ```suggestion
2. When you want to check for updates, just run `git pull orign` in the `cht-core` directory.
```

I think this is technically unnecessary since usually markdown rendering engines are smart enough to make the numbering sequential, but can't hurt.

Author: thanks!

#### Launching

Run the script with:

```
./cht-docker-compose.sh
```

The first time you run, you will be prompted to create a new project.  Here's what that looks like:

```shell
./cht-docker-compose.sh
Would you like to initialize a new project [y/N]? y
How do you want to name the project? 4 OH The First
Downloading compose files ... done 

Creating network "4_oh_the_first-cht-net" with the default driver
Creating my_first_cht_project-dir_cht-upgrade-service_1 ... done
Starting project "4_oh_the_first". First run takes a while. Will try for up to five minutes........

 -------------------------------------------------------- 

  Success! "4_oh_the_first" is set up:

    https://127-0-0-1.my.local-ip.co:10444/ (CHT)
    https://127-0-0-1.my.local-ip.co:10444/_utils/ (Fauxton)

    Login: medic
    Password: password

 -------------------------------------------------------- 

 Have a great day! 
```

If you have many existing projects, you can specify them to launch them directly. If you had a project called `4_oh_the_first` you would run:

```shell
./cht-docker-compose.sh 4_oh_the_first.env
```

#### Stopping

> When you're done with a project, it's good to stop all the containers to reduce load on your computer.  Do this by specifying the project and the `down` command. Using our existing example `4_oh_the_first` project, you would call:

Reviewer: Would be nice to note that all the data on the instance will be retained and available the next time it is started.

Author: ah - good call! put that in cf229f6

```shell
./cht-docker-compose.sh 4_oh_the_first.env down
```

#### Destroying

> When you want to fully remove all contents of a project by deleting them, specifying the project and the `destroy` command. Using our existing example `4_oh_the_first` project, you would call:

Reviewer: Would be nice to note that all the data from the instance will be LOST.

Author: good call! fixed in d593631
